contrib/endpoints/src/api_manager/check_security_rules_test.cc

```diff
@@ -0,0 +1,413 @@
+        Config::Create(raw_env_, service_config, server_config);
+
+    ASSERT_TRUE(config != nullptr);
+
+    service_context_ = std::make_shared<context::ServiceContext>(
+        std::move(env), std::move(config));
+
+    ASSERT_TRUE(service_context_.get() != nullptr);
+
+    std::unique_ptr<MockRequest> request(
+        new ::testing::NiceMock<MockRequest>());
+    raw_request = request.get();
+
+    request_context_ = std::make_shared<context::RequestContext>(
+        service_context_, std::move(request));
+    EXPECT_CALL(*raw_env_, DoRunHTTPRequest(_)).Times(0);
+  }
+
+  MockApiManagerEnvironment *raw_env_;
+  MockRequest *raw_request;
+  std::shared_ptr<context::RequestContext> request_context_;
+  std::shared_ptr<context::ServiceContext> service_context_;
+};
+
+TEST_P(CheckDisableSecurityRulesTest, CheckAuthzDisabled) {
+  CheckSecurityRules(request_context_,
+                     [](Status status) { ASSERT_TRUE(status.ok()); });
+}
+
+INSTANTIATE_TEST_CASE_P(ConfigToDisableFirebaseRulesCheck,
+                        CheckDisableSecurityRulesTest,
+                        testing::Values(GetConfigWithNoAuth(),
+                                        GetConfigWithoutApis(),
+                                        GetConfigWithoutServer()));
+
+class CheckSecurityRulesTest : public ::testing::Test {
+ public:
+  void SetUp(std::string service_config, std::string server_config) {
+    std::unique_ptr<MockApiManagerEnvironment> env(
+        new ::testing::NiceMock<MockApiManagerEnvironment>());
+    raw_env_ = env.get();
+
+    std::unique_ptr<Config> config =
+        Config::Create(raw_env_, service_config, server_config);
+    ASSERT_TRUE(config != nullptr);
+
+    service_context_ = std::make_shared<context::ServiceContext>(
+        std::move(env), std::move(config));
+
+    ASSERT_TRUE(service_context_.get() != nullptr);
+
+    std::unique_ptr<MockRequest> request(
+        new ::testing::NiceMock<MockRequest>());
+    raw_request_ = request.get();
+
+    ON_CALL(*raw_request_, GetRequestHTTPMethod())
+        .WillByDefault(Return(std::string("GET")));
+
+    ON_CALL(*raw_request_, GetRequestPath())
+        .WillByDefault(Return(std::string("/ListShelves")));
+
+    request_context_ = std::make_shared<context::RequestContext>(
+        service_context_, std::move(request));
+    release_url_ =
+        "https://myfirebaseserver.com/v1/projects/myfirebaseapp/"
+        "releases/myfirebaseapp.appspot.com:v1";
+
+    ruleset_test_url_ =
+        "https://myfirebaseserver.com/v1"
+        "/projects/myfirebaseapp/rulesets/99045fc0-a5e4-47e2-a665-f88593594b6b"
+        ":test?alt=json";
+  }
+
+  MockApiManagerEnvironment *raw_env_;
+  MockRequest *raw_request_;
+  std::shared_ptr<context::RequestContext> request_context_;
+  std::shared_ptr<context::ServiceContext> service_context_;
+  std::string release_url_;
+  std::string ruleset_test_url_;
+};
+
+TEST_F(CheckSecurityRulesTest, CheckAuthzFailGetRelease) {
```

Comment:
Can you add comments for each test, explaining what the test setup and expectation?